src/views/LocationDetail.vue

```diff
@@ -62,13 +89,43 @@ export default {
       locationPriceTotal: null,
       locationPricePage: 0,
       loading: false,
+      // filter & order
+      priceFilter: '',
+      priceFilterList: constants.PRICE_FILTER_LIST,
+      priceOrder: constants.PRICE_ORDER_BY_LIST[1].key,
+      priceOrderList: constants.PRICE_ORDER_BY_LIST,
     }
   },
   mounted() {
+    this.priceFilter = this.$route.query[constants.FILTER_PARAM] || this.priceFilter
+    this.priceOrder = this.$route.query[constants.ORDER_BY_PARAM] || this.priceOrder
     this.getLocation(),
     this.getLocationPrices()
   },
+  computed: {
+    locationId() {
+      return this.$route.params.id
+    },
+    getCurrentPriceOrderIcon() {
+      let currentPriceOrder = this.priceOrderList.find(o => o.key === this.priceOrder)
+      return currentPriceOrder ? currentPriceOrder.icon : ''
+    },
+    getPricesParams() {
+      let defaultParams = { location_id: this.locationId, order_by: `${this.priceOrder}`, page: this.locationPricePage }
+      if (this.priceFilter === 'show_last_month') {
+        let oneMonthAgo = new Date()
+        oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1)
+        defaultParams['date__gte'] = oneMonthAgo.toISOString().split('T')[0]
```

Comment:
in the code elsewhere we used `toISOString().substring(0, 10)`. better ?

Reply:
Tested on a 1000000 loop
split: 318.093994140625 ms
substring: 3.641845703125 ms
split: 264.632080078125 ms
substring: 0.674072265625 ms

substring is indeed faster (300 nano second compared to 1ns). But both are still super fast. I guess it is down to readability. Both are fine by me

Reply:
Wasn't thinking about speed. Just uniformity 👍